nx/include/switch/kernel/uevent.h

```diff
@@ -0,0 +1,23 @@
+// Copyright 2018 plutoo
+#pragma once
+#include "kernel/mutex.h"
+#include "kernel/waiter.h"
+
+typedef struct UsermodeEvent UsermodeEvent;
+
+struct UsermodeEvent
+{
+    Mutex mutex;
+    WaiterNode waiter_list;
+    bool signal;
+    bool auto_clear;
+};
+
+void ueventCreate(UsermodeEvent* e, bool auto_clear);
+void ueventClear(UsermodeEvent* e);
+void ueventSignal(UsermodeEvent* e);
+
+// Internal methods (do not use!):
```

Comment:
Internal methods should not be in public header files. These should be in an internal header inside source/kernel/.

Reply:
We don't use this pattern elsewhere

Reply:
We do, it's internal.h.